packages/eas-cli/src/commands/env/init.ts

```diff
@@ -0,0 +1,231 @@
+import spawnAsync from '@expo/spawn-async';
+import { appendFile, pathExists, readFile, writeFile } from 'fs-extra';
+import os from 'os';
+import path from 'path';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { EASNonInteractiveFlag } from '../../commandUtils/flags';
+import Log from '../../log';
+import { confirmAsync } from '../../prompts';
+
+const ENVRC_TEMPLATE =
+  'dotenv_if_exists .env;\ndotenv_if_exists .env.eas.local;\ndotenv_if_exists .env.local;\n';
+
+export default class EnvironmentVariableInit extends EasCommand {
+  static override description = 'setup environment variables';
+
+  static override hidden = true;
+
+  static override flags = {
+    ...EASNonInteractiveFlag,
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectDir,
+  };
+
+  async runAsync(): Promise<void> {
+    const {
+      flags: { 'non-interactive': nonInteractive },
+    } = await this.parse(EnvironmentVariableInit);
+
+    if (nonInteractive) {
+      throw new Error("Non-interactive mode is not supported for 'eas env:init'");
+    }
+
+    const { projectDir } = await this.getContextAsync(EnvironmentVariableInit, {
+      nonInteractive,
+    });
+
+    await this.ensureDirenvInstalledAsync(projectDir);
+    await this.setupEnvrcFileAsync(projectDir);
+    await this.addDirenvHookToShellConfigAsync();
+    await this.addToGitIgnoreAsync(projectDir);
+    Log.log('Running direnv allow...');
+    await spawnAsync('direnv', ['allow'], { cwd: projectDir, stdio: 'inherit' });
+  }
+
+  private async addDirenvHookToShellConfigAsync(): Promise<void> {
+    const direnvConfig = this.getShellDirenvConfig();
+
+    if (direnvConfig && (await pathExists(direnvConfig.shellConfigPath))) {
+      const { shellConfigPath, direnvHookCmd, direnvInitCmd } = direnvConfig;
+
+      const confirm = await confirmAsync({
+        message: `Do you want to add the direnv hook to ${shellConfigPath}?`,
+      });
+
+      if (!confirm) {
+        Log.log('Skipping adding the direnv hook to the shell config');
+        Log.log('You may need to add the direnv hook to your shell config manually.');
+        Log.log('Learn more: https://direnv.net/docs/hook.html');
+        return;
```

Comment:
```suggestion
        throw new Error
```
to make it exit with non-zero code in such case